### PR TITLE
kubeadm: use the phase name to unify the logs

### PR DESCRIPTION
**Pull request by chendave**
- The method like `InitCfg()` could be called by phases other that the `preflight`, it's confusing to say it is `preflight` while it's actually try to join a control node.

e.g.
```
kubeadm join phase control-plane-join mark-control-plane --control-plane -v 5
...
I0322 10:21:20.981886  556656 join.go:539] [preflight] Fetching init configuration
I0322 10:21:20.981915  556656 join.go:585] [preflight] Retrieving KubeConfig objects
...
```

- It is better to say the current phase name instead of it's parent phase name, that could be easier to indentify where we are.

e.g. If it's the `update-status` sub-phase, it would be better to say `update-status` instead of `control-plane-join` in the log.

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig cluster-lifecycle

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
@chendave 

i am not so sure about this change. it might be better for us to keep the [] prefixes static. e.g. "preflight" might not be very accurate, but at least it signifies an action before the actual code kicks in.

interesting to see what other think about this too.

File: cmd/kubeadm/app/cmd/phases/init/uploadconfig.go [hunk 1]
```diff
@@ -103,42 +103,42 @@ func getUploadConfigPhaseFlags() []string {
 }
 
 // runUploadKubeadmConfig uploads the kubeadm configuration to a ConfigMap
-func runUploadKubeadmConfig(c workflow.RunData) error {
-	cfg, client, _, err := getUploadConfigData(c)
+func runUploadKubeadmConfig(c workflow.RunData, phase string) error {
+	cfg, client, _, err := getUploadConfigData(c, phase)
```
Comment (pacoxu) [inline cmd/kubeadm/app/cmd/phases/init/uploadconfig.go:107]: The phase here can be `kubelet`, `kubeadm`, and `all`.

```
kubeconfig                   Generate all kubeconfig files necessary to establish the control plane and the admin kubeconfig file
  /kubelet                     Generate a kubeconfig file for the kubelet to use *only* for cluster bootstrapping purposes
...
upload-config                Upload the kubeadm and kubelet configuration to a ConfigMap
  /kubeadm                     Upload the kubeadm ClusterConfiguration to a ConfigMap
  /kubelet                     Upload the kubelet component config to a ConfigMap
```

There will be two `[kubelet]`, one for kubeconfig and one for upload-config.

Similarly, `apiserver` and `controller-manager `, `scheduler` also may be for different phase.

```
certs                        Certificate generation
  /apiserver                   Generate the certificate for serving the Kubernetes API
kubeconfig                   Generate all kubeconfig files necessary to establish the control plane and the admin kubeconfig file
  /kubelet                     Generate a kubeconfig file for the kubelet to use *only* for cluster bootstrapping purposes
  /controller-manager          Generate a kubeconfig file for the controller manager to use
  /scheduler                   Generate a kubeconfig file for the scheduler to use
control-plane                Generate all static Pod manifest files necessary to establish the control plane
  /apiserver                   Generates the kube-apiserver static Pod manifest
  /controller-manager          Generates the kube-controller-manager static Pod manifest
  /scheduler                   Generates the kube-scheduler static Pod manifest
upload-config                Upload the kubeadm and kubelet configuration to a ConfigMap
  /kubelet                     Upload the kubelet component config to a ConfigMap
```

File: cmd/kubeadm/app/cmd/join.go [hunk 2]
```diff
@@ -526,22 +526,22 @@ func (j *joinData) TLSBootstrapCfg() (*clientcmdapi.Config, error) {
 	if j.tlsBootstrapCfg != nil {
 		return j.tlsBootstrapCfg, nil
 	}
-	klog.V(1).Infoln("[preflight] Discovering cluster-info")
```
[the remark this answers is not in the document]
Comment (chendave) [reply cmd/kubeadm/app/cmd/join.go:529]: but how about the phase like this: `kubeadm join phase kubelet-start`? is that okay to say this is in the phase `preflight` in the log?

File: cmd/kubeadm/app/cmd/phases/workflow/phase.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ type Phase struct {
 	// Run defines a function implementing the phase action.
 	// It is recommended to implent type assertion, e.g. using golang type switch,
```
[the remark this answers is not in the document]
Comment (chendave) [reply cmd/kubeadm/app/cmd/phases/workflow/phase.go:57]: Good catch, will fix it.

File: cmd/kubeadm/app/cmd/phases/init/uploadconfig.go (hunk 1 above)
Comment (chendave) [reply cmd/kubeadm/app/cmd/phases/init/uploadconfig.go:107]: I'd hope to make the log as precise as possible with minimal cost, your comments here is valid and nice catch, but I don't make it over-complicated, thoughts?

**Comment by chendave** [comment]
@pacoxu thanks for the review, pls check the inline comments for the reply.

File: cmd/kubeadm/app/cmd/phases/init/uploadconfig.go (hunk 1 above)
Comment (pacoxu) [reply cmd/kubeadm/app/cmd/phases/init/uploadconfig.go:107]: - Before, only `parent-phase` will be shown. The log details may show what phase it is.
- After this change, the `sub-phase` is a priority over the `parent-phase`. It is more precise but sometimes dup with another sub-phase.

For instance, before
```
[certs] Using certificateDir folder "/etc/kubernetes/tmp/kubeadm-init-dryrun443308956"
[certs] Using existing ca certificate authority
[certs] Using existing apiserver certificate and key on disk
[certs] Using existing apiserver-kubelet-client certificate and key on disk
[certs] Using existing front-proxy-ca certificate authority
[certs] Using existing front-proxy-client certificate and key on disk
[certs] Using existing etcd/ca certificate authority
[certs] Using existing etcd/server certificate and key on disk
[certs] Using existing etcd/peer certificate and key on disk
[certs] Using existing etcd/healthcheck-client certificate and key on disk
[certs] Using existing apiserver-etcd-client certificate and key on disk
[certs] Generating "sa" key and public key
```
after 
```
[certs] Using certificateDir folder "/etc/kubernetes/tmp/kubeadm-init-dryrun443308956"
[ca] Using existing ca certificate authority
[apiserver] Using existing apiserver certificate and key on disk
[apiserver-kubelet-client] Using existing apiserver-kubelet-client certificate and key on disk
[front-proxy-ca] Using existing front-proxy-ca certificate authority
[front-proxy-client] Using existing front-proxy-client certificate and key on disk
[etcd-ca] Using existing etcd/ca certificate authority
[etcd-server] Using existing etcd/server certificate and key on disk
[etcd-peer] Using existing etcd/peer certificate and key on disk
[etcd-healthcheck-client] Using existing etcd/healthcheck-client certificate and key on disk
[apiserver-etcd-client] Using existing apiserver-etcd-client certificate and key on disk
[ca] Generating "sa" key and public key
```


I prefer the current style in this usecase.

Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/init/uploadconfig.go:107]: > I prefer the current style in this usecase.

+1 i agree. we should try to keep it simple. if some `[  ... ]` prefixes need adjustment we can fix them, though.

Comment (SataQiu) [reply cmd/kubeadm/app/cmd/phases/init/uploadconfig.go:107]: Also prefer the keep original style here. There is no need to print the specific stages as the prefix, and from the content we can determine what we are doing.

**Comment by chendave** [comment]
based on the review comments.

/close

**Pull request closed without merging**